projects/dendrites/permutedMNIST/experiments/base.py

```diff
@@ -93,8 +106,170 @@ class PermutedMNISTExperiment(mixins.RezeroWeights,
+
+    model_args=dict(
+        input_size=784,
+        output_size=10,
+        hidden_sizes=[2048, 2048],
+        num_segments=10,    # NUM_TASKS
+        dim_context=1024,
+        kw=True,
+        dendrite_weight_sparsity=0.0,
+        weight_sparsity=tune.sample_from(
+            lambda spec: np.random.choice([0.95, 0.90, 0.8])),
+    ),
+
+    num_tasks=10,            # NUM_TASKS
+    num_classes=100,         # 10 * NUM_TASKS
+    seed=tune.sample_from(lambda spec: np.random.randint(2, 10000)),
+    num_samples=20,
+
+    optimizer_class=tune.grid_search([torch.optim.Adam, torch.optim.SGD]),
+    optimizer_args=dict(
+        lr=tune.sample_from(
+            lambda spec: np.random.choice([0.01, 0.005, 0.001, 0.0005])),
+    ),
+)
+
+# 10 tasks. Currently several variables depend on tasks and are hardcoded in config
+BASE_10_SPARSITY_SEARCH = deepcopy(BASE_10_TASKS)
+BASE_10_SPARSITY_SEARCH.update(
+    model_args=dict(
+        input_size=784,
+        output_size=10,
+        hidden_sizes=[2048, 2048],
+        num_segments=10,    # NUM_TASKS
+        dim_context=1024,
+        kw=True,
+        kw_percent_on=tune.sample_from(
+            lambda spec: np.random.choice([0.05, 0.1, 0.2, 0.0])),
+        dendrite_weight_sparsity=0.0,
+        weight_sparsity=tune.sample_from(
+            lambda spec: np.random.choice([0.90, 0.8, 0.7, 0.5, 0.25, 0.0])),
+    ),
+
+    epochs=tune.sample_from(lambda spec: np.random.randint(1, 3)),
+    num_samples=5,
+
+    optimizer_class=torch.optim.Adam,
+    optimizer_args=dict(
+        lr=tune.sample_from(lambda spec: np.random.choice([0.001, 0.0005])),
+    ),
+)
+
+# 50 tasks. Currently several variables depend on tasks and are hardcoded in config
+BASE_50_SPARSITY_SEARCH = deepcopy(BASE_10_SPARSITY_SEARCH)
+BASE_50_SPARSITY_SEARCH.update(
+    dataset_args=dict(
+        num_tasks=50,        # NUM_TASKS
+        root=os.path.expanduser("~/nta/results/data/"),
+        dim_context=1024,
+        seed=42,
+        download=False,  # Change to True if running for the first time
+    ),
+
+    model_args=dict(
+        input_size=784,
+        output_size=10,
+        hidden_sizes=[2048, 2048],
+        num_segments=50,    # NUM_TASKS
+        dim_context=1024,
+        kw=True,
+        kw_percent_on=tune.sample_from(
+            lambda spec: np.random.choice([0.05, 0.1, 0.2, 0.0])),
+        dendrite_weight_sparsity=0.0,
+        weight_sparsity=tune.sample_from(
+            lambda spec: np.random.choice([0.90, 0.8, 0.5, 0.0])),
+    ),
+
+    tasks_to_validate=(0, 1, 40, 49, 50),  # Tasks on which to run validate
+    epochs=1,
+    num_samples=8,
+    num_tasks=50,  # NUM_TASKS
+    num_classes=500,  # 10 * NUM_TASKS
+
+    optimizer_class=torch.optim.Adam,
+    optimizer_args=dict(
+        lr=tune.sample_from(lambda spec: np.random.choice([0.001, 0.0005])),
+    ),
+)
+
+# Centroid method for inferring contexts: 10 permutedMNIST tasks
+CENTROID_10 = dict(
```

Comment:
I think we should move all the basic centroid experiments into its own experiments/centroid.py file.

Reply:
Done.